File: app/src/main/java/com/azure/minitwitter/data/TweetViewModel.java
package com.azure.minitwitter.data;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.arch.lifecycle.LiveData;
import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.app.AppCompatActivity;

import com.azure.minitwitter.retrofit.response.Tweet;
import com.azure.minitwitter.ui.BottomModalTweetFragment;

import java.util.List;

public class TweetViewModel extends AndroidViewModel {

    private TweetRepository tweetRepository;
    private LiveData<List<Tweet>> tweets;
    private LiveData<List<Tweet>> favTweets;

    public TweetViewModel(@NonNull Application application) {
        super(application);
        tweetRepository = new TweetRepository();
        tweets = tweetRepository.getAllTweets();
    }

    public LiveData<List<Tweet>> getTweets() {
        return tweets;
    }

    public LiveData<List<Tweet>> getFavTweets() {
        favTweets = tweetRepository.getFavTweets();
        return favTweets;
    }

    public LiveData<List<Tweet>> getNewTweets() {
        tweets = tweetRepository.getAllTweets();
        return tweets;
    }

    public LiveData<List<Tweet>> getNewFavTweets() {
        getNewTweets();
        return getFavTweets();
    }

    public void insertTweet(String message){
        tweetRepository.createTweet(message);
    }

    public void likeTweet(int idTweet){
        tweetRepository.likeTweet(idTweet);
    }

    public void deleteTweet(int idTweet) {
        tweetRepository.deleteTweet(idTweet);
    }

    public void openDialogTweetMenu(Context ctx, int idTweet){
        BottomModalTweetFragment dialogTweet = BottomModalTweetFragment.newInstance(idTweet);
        dialogTweet.show(((AppCompatActivity)ctx).getSupportFragmentManager(),"BottomModalTweetFragment");
    }
}

File: app/src/main/java/com/azure/minitwitter/data/TweetViewModel.kt
package com.azure.minitwitter.data

import android.app.Application
import android.arch.lifecycle.AndroidViewModel
import android.arch.lifecycle.LiveData
import android.arch.lifecycle.MutableLiveData
import android.content.Context
import android.support.v7.app.AppCompatActivity
import com.azure.minitwitter.retrofit.response.Tweet
import com.azure.minitwitter.ui.tweets.BottomModalTweetFragment

class TweetViewModel(application: Application) : AndroidViewModel(application){

    private val tweetRepository: TweetRepository = TweetRepository()

    private var tweets = tweetRepository.getAllTweets()
//        get() = tweets

    private val favTweets : LiveData<List<Tweet>> = tweetRepository.getFavTweets()
//        get() = favTweets

    fun getNewTweets(): MutableLiveData<List<Tweet>> {
        tweets = tweetRepository.getAllTweets()
        return tweets
    }

    fun getNewFavTweets(): LiveData<List<Tweet>> {
        getNewTweets()
        return favTweets
    }

    fun insertTweet(message: String): Unit = tweetRepository.createTweet(message)

    fun likeTweet(idTweet: Int): Unit  = tweetRepository.likeTweet(idTweet)

    fun deleteTweet(idTweet: Int): Unit = tweetRepository.deleteTweet(idTweet)

    fun openDialogTweetMenu(ctx: Context, idTweet: Int){
        val dialogTweet: BottomModalTweetFragment = BottomModalTweetFragment.newInstance(idTweet)
        dialogTweet.show((ctx as AppCompatActivity).supportFragmentManager, "BottomModalTweetFragment")
    }
}